concepts/80 Troubleshooting/05 DataGrid and TreeList Issues/10 Gray Boxes Appear After Data Loading/00 Gray Boxes Appear After Data Loading.md

```diff
@@ -0,0 +1,12 @@
+In certain cases, gray boxes may remain visible after DataGrid or TreeList finishes data loading:
+
+![DataGrid with Gray Boxes Visible](/images/DataGrid/datagrid-gray-boxes.png)
+
+These gray boxes are row placeholders that appear when the component takes time to load and render rows. These placeholders can remain visible due to the following reasons:
+
+- The component's size does not match the container size.
+- An error occurs during data loading.
+- Loaded data is incorrect (has wrong structure, missing required fields, the number of loaded items is not expected, etc.)
+- Unexpected external interactions (the component is re-rendering, the container has been resized, saved component state is corrupted, etc.)
```

Comment:
First. We should stick to the same tense. A simple one, preferably.

Second. "state" is countable, which requires an article from us.

```suggestion
- Unexpected external interactions (when the component is re-rendering, the container is resized, the saved component state is corrupted, etc.)
```